CRM/Admin/Form/SettingTrait.php

```diff
@@ -227,7 +228,14 @@ protected function addFieldsDefinedInSettingsMetadata() {
           $this->addRadio($setting, $props['title'], [1 => 'Yes', 0 => 'No'], CRM_Utils_Array::value('html_attributes', $props), '&nbsp;&nbsp;');
         }
         elseif ($add === 'add') {
-          $this->add($props['html_type'], $setting, $props['title'], $options);
+          if ($props['html_type'] === 'datepicker') {
+            if (CRM_Utils_Array::value('minDate', $props['html_extra'])) {
```

Comment:
FYI I try to avoid using this function when the return value is immediately cast to boolean, as it's slower and less readable than

```suggestion
            if (!empty($props['html_extra']['minDate'])) {
```